ledger/participant-integration-api/src/main/scala/platform/store/Conversions.scala

```diff
@@ -142,7 +144,56 @@ private[platform] object Conversions {
   }
 
   object DefaultImplicitArrayColumn {
-    val default = Column.of[Array[String]]
+    val defaultString: Column[Array[String]] = Column.of[Array[String]]
+    val defaultInt: Column[Array[Int]] = Column.of[Array[Int]]
+  }
+
+  object ArrayColumnToIntArray {
+    implicit val arrayColumnToIntArray: Column[Array[Int]] = nonNull { (value, meta) =>
+      DefaultImplicitArrayColumn.defaultInt(value, meta) match {
+        case Right(value) => Right(value)
+        case Left(_) =>
+          val MetaDataItem(qualified, _, _) = meta
+          value match {
+            case someArray: Array[_] =>
+              Try(
+                someArray.view.map {
+                  case i: Int => i
+                  case null =>
+                    throw new Exception(s"Cannot convert object array element null to Int")
```

Comment:
Let's make sure they are handled properly with the new error codes.
```suggestion
                    throw new SQLNonTransientException(s"Cannot convert object array element null to Int")
```